src/app/server/Server.cpp

```diff
@@ -572,8 +574,14 @@ CHIP_ERROR AddTestPairing()
     VerifyOrExit(adminInfo != nullptr, err = CHIP_ERROR_NO_MEMORY);
 
     adminInfo->SetNodeId(chip::kTestDeviceNodeId);
+    gTestPairing.ToSerializable(serializedTestSession);
+
+    testSession = chip::Platform::New<PASESession>();
+    testSession->FromSerializable(serializedTestSession);
     SuccessOrExit(err = gSessions.NewPairing(Optional<PeerAddress>{ PeerAddress::Uninitialized() }, chip::kTestControllerNodeId,
-                                             &gTestPairing, SecureSessionMgr::PairingDirection::kResponder, gNextAvailableAdminId));
+                                             testSession, SecureSessionMgr::PairingDirection::kResponder, gNextAvailableAdminId));
+    testSession->Clear();
+    chip::Platform::Delete(testSession);
```

Comment:
this won't be called on failure. Move after `exit:` and add `if (testSession)` ?

Reply:
> this won't be called on failure. Move after `exit:` and add `if (testSession)` ?

Urgh. For some reasons I missed the call to `SuccessOrExit`. I'm used to the pattern where: 
```
  err = ...;
  SuccessOrExit(err);
```

but it is a one liner here :/

Thanks for catching that!